### DISCUSSION TITLE
Search should start from the current position

### DISCUSSION THREAD

**Discussion by italomaia-bk** [helix-editor/helix#10494] [General]
### Summary

Search does not match text that starts at the cursor position.

### Reproduction Steps

 

For the text:
```
$ hello
$ world
```
Given cursor is just behind the first dollar character, if I have `\$ ` in my `*` register and type "n" to find the next match, the beginning of line 2 will be the first match.
This is particularly problematic if you were in **select mode** and type "d" because it will delete the current cursor position AND the match in line two, which is not what I would expect. 

### Helix log

nothing to see

### Platform

Linux

### Terminal Emulator

xfce4-terminal

### Installation Method

snap

### Helix Version

helix 24.3 (2cadec0b)

**Comment by pascalkuthe** [comment]
Are you using a bar cursor? `n` will jump to the second dollar only if it's *on` the first dollar sign.  This is intentional otherwise pressing n multiple times wouldn't work

**Comment by italomaia-bk** [comment]
I am using the bar cursor BUT the problem does not happen with other characters. Could the fact dollar is escaped influence here?

```
|10
20
```
`|` <- position of the curosr
`*` = "10"
If I type "n", it finds 10. For the following scenario:
```
|$ 10
$ 20
```
`|` <- position of the curosr
`*` = "\$ "
If I type "n", it does not find text in the first line.

**Reply by pascalkuthe** [reply]
the bar cursor is a visual gimmick the cursor is always a block cursor. `*n` is fundamentally supposed to always jump the cursor. This has nothing to do with `$` it works like that for any character.

The cursor is not before the `$` it is *on* the dollar and the search will find the next match from the end of the selection. I would strongly advise against using the bar cursor outside of insertmode (or really in general). It does not reflect our selection semantics

**Reply by italomaia-bk** [reply]
My bad, the cursor was in block mode, not bar. 
The bar cursor does not behave well in insert mode (in my opinion) because commands like **move_next_word_end** will place the cursor behind the last character, which can be confusing.
